### PR TITLE
HDDS-2876. Consolidate ObjectID and UpdateID from Info objects into one class.

### PR DESCRIPTION
**Pull request by hanishakoneru**
## What changes were proposed in this pull request?

We use ObjectID and BucketID in OMVolumeArgs, OMBucketInfo, OMKeyInfo and OMMultipartKeyInfo. We can consolidate by having these Info objects extend a "WithObjectID" class which can host the common fields - objectID and updateID.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-2876

## How was this patch tested?

Code consolidation. Separate Tests not required.

### CODE REVIEW

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java [hunk 1]
```diff
@@ -463,6 +433,19 @@ public static OmKeyInfo getFromProtobuf(KeyInfo keyInfo) {
     return builder.build();
   }
 
+  @Override
+  public String getObjectInfo() {
+    return "OMVolumeArgs{" +
```
Comment (adoroszlai) [inline hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java]: Shouldn't this be `OmKeyInfo`?

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmBucketInfo.java [hunk 2]
```diff
@@ -452,6 +401,17 @@ public static OmBucketInfo getFromProtobuf(BucketInfo bucketInfo) {
     return obib.build();
   }
 
+  @Override
+  public String getObjectInfo() {
+    return "OMVolumeArgs{" +
```
Comment (adoroszlai) [inline hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmBucketInfo.java]: Shouldn't this be `OmBucketInfo`?

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java (hunk 1 above)
Comment (hanishakoneru) [reply hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java]: Fixed.

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmBucketInfo.java (hunk 2 above)
Comment (hanishakoneru) [reply hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmBucketInfo.java]: Fixed.

**Comment by hanishakoneru** [comment]
Thanks for the review @adoroszlai . Fixed the object names.

**Comment by elek** [comment]
Fresh build is passed: 
https://github.com/apache/hadoop-ozone/commit/a7bc81ed113195a5c72871ad89cbb7308e307ac7

Committing now with a minor conflict resolution:

```
         OmKeyInfo dbKeyInfo = omMetadataManager.getKeyTable().get(ozoneKey);
         if (dbKeyInfo != null) {
-          if (isReplay(ozoneManager, dbKeyInfo.getUpdateID(), trxnLogIndex)) {
+          if (isReplay(ozoneManager, dbKeyInfo, trxnLogIndex)) {
             // This transaction is a replay. Send replay response.
             throw new OMReplayException();
```

**Comment by hanishakoneru** [comment]
Thank you @elek and @adoroszlai for reviewing and merging the PR.

**Pull request closed without merging**